llvm/include/llvm/IR/IntrinsicsDPU.td

```diff
@@ -11,8 +11,12 @@
 //
 //===----------------------------------------------------------------------===//
 
+def llvm_mram_ptr_ty : LLVMQualPointerType<llvm_i8_ty, 255>;
```

Comment:
Does `llvm_i8_ty` will be an issue for a i32* argument (for example)? Or is it the default "void *" type?

Reply:
I think that the type here are LLVM type, not C type. I did try to declare a equivalent of a ``void *`` type, but it failed, but I'm not sure why it failed at that time, it might have been for another reason.

Anyway, the front-end (clang), wants a ``__mram_ptr void *``, so if you give it a ``__mram_ptr int *``, it should at least give a warning...
I'll try to make some tested to have clearer idea on this topic.

Reply:
Changed to use some kind of ``void *`` type.